spmd/tensor/ops/math_ops.py

```diff
@@ -211,3 +216,35 @@ def reduction_rule(op_schema: OpSchema) -> OutputSharding:
 
 for reduction_op in reduction_ops:
     DTensor._op_to_rules[reduction_op] = reduction_rule
+
+
+@register_prop_rule("aten._softmax.default")
+def softmax_rule(op_schema: OpSchema) -> OutputSharding:
+    input_spec = op_schema.args_spec[0]
+    dim_map = input_spec.dim_map
+    softmax_dim = cast(
+        int, op_schema.args_schema[len(op_schema.args_spec)]
+    )  # Is it better to put it into kwargs? e.g. op_schema.kwargs_schema['dim']
```

Comment:
https://github.com/pytorch/pytorch/blob/1cad744694d7feb7c55e5f4ff4a6ae749686bfb5/aten/src/ATen/native/native_functions.yaml#L4721

softmax is only having positional argument for `dim`, so that's just keep positional arg